logback-contrib/json/core/src/main/java/ch/qos/logback/contrib/json/JsonLayoutBase.java

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright (C) 2024, LINAGORA. All rights reserved.
+ *
+ * This program and the accompanying materials are licensed under
+ * the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ * 
+ * This work is based on the work found at
+ * https://github.com/qos-ch/logback-contrib
+ * authored by the logback-contrib developers.
+ */
+package ch.qos.logback.contrib.json;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Map;
+import java.util.TimeZone;
+
+import ch.qos.logback.core.CoreConstants;
+import ch.qos.logback.core.LayoutBase;
+
+/**
+ * @author Les Hazlewood
+ * @author Pierre Queinnec
+ * @author Espen A. Fossen
+ * @since 0.1
+ */
+public abstract class JsonLayoutBase<E> extends LayoutBase<E> {
+
+    public static final String CONTENT_TYPE = "application/json";
+
+    protected boolean includeTimestamp;
+    protected String timestampFormat;
+    protected String timestampFormatTimezoneId;
+    protected boolean appendLineSeparator;
+
+    protected JsonFormatter jsonFormatter;
+
+    public JsonLayoutBase() {
+        this.includeTimestamp = true;
+        this.appendLineSeparator = false;
+    }
+
+    @Override
+    public String doLayout(E event) {
+        Map map = toJsonMap(event);
+        if (map == null || map.isEmpty()) {
+            return null;
+        }
+        String result = getStringFromFormatter(map);
+        return isAppendLineSeparator() ? result + CoreConstants.LINE_SEPARATOR : result;
+    }
+
+    private String getStringFromFormatter(Map map) {
+        JsonFormatter formatter = getJsonFormatter();
+        if (formatter == null) {
+            addError("JsonFormatter has not been configured on JsonLayout instance " + getClass().getName() + ".  Defaulting to map.toString().");
+            return map.toString();
+        }
+
+        try {
+            return formatter.toJsonString(map);
+        } catch (Exception e) {
+            addError("JsonFormatter failed.  Defaulting to map.toString().  Message: " + e.getMessage(), e);
+            return map.toString();
+        }
+    }
+
+    protected String formatTimestamp(long timestamp) {
+        if (this.timestampFormat == null || timestamp < 0) {
+            return String.valueOf(timestamp);
+        }
+        Date date = new Date(timestamp);
+        DateFormat format = createDateFormat(this.timestampFormat);
+
+        if (this.timestampFormatTimezoneId != null) {
+            TimeZone tz = TimeZone.getTimeZone(this.timestampFormatTimezoneId);
+            format.setTimeZone(tz);
+        }
```

Comment:
Please let's rewrite this bit with java 8 time:

 - Use Instant instead of Date
 - Use DateTimeFormatter
 
 Why? Because SimpleDateTime is crazy inefficient!
 
 CF flame graphs:
 
 As shown here rougthly 50% of the logging time is spent on simple date format...
 
![image](https://github.com/user-attachments/assets/056357d3-3031-47fa-9f5b-03762f518e8d)

(that's why we are forking...)